pkg/operator/sync.go

```diff
@@ -424,6 +425,16 @@ func (optr *Operator) syncMachineConfigDaemon(config *renderConfig) error {
 		return err
 	}
 
+	etcdQuorumGuardBytes, err := renderAsset(config, "manifests/etcdquorumguard_deployment.yaml")
+	if err != nil {
+		return err
+	}
+	etcdQuorumGuard := resourceread.ReadDeploymentV1OrDie(etcdQuorumGuardBytes)
```

Comment:
looks like this function can panic, maybe better use `resourceread.ReadDeploymentV1` and return an error?

Reply:
This is expected - operator should crash if the rendered version is not a valid deployment. Throwing a panic here is probably too harsh, but that's what other functions here do